### DISCUSSION TITLE
How can I change the CPU metrics to reflect a percentage of my total cpu capacity, not whatever it's based off of currently?

### DISCUSSION THREAD

**Discussion by philocalyst** [ClementTsang/bottom#1563] [Q&A] [answered]
I've noticed that the cpu usage display will be much higher than the current average usage. I don't know how to configure bottom to display it out of my total CPU capacity like in other programs (like btop)

Thank you!

**Comment by ClementTsang** [comment]
Sorry, which widget is this for? The CPU widget? Or the processes widget? And can you show an example of what you mean? As bottom _should_ be displaying things like process CPU usage based on total CPU by default.

**Reply by philocalyst** [reply]
Apologies, I keep forgetting to provide all context for my issues. This is for the process widget, the CPU widget is reporting values accurately as far as I can tell in my comparison with the other monitors I have. A screenshot is shown below, where the wezterm GUI is taking ~4x more of my CPU capacity than the current average. That is just the easiest example I was able to produce.
<img width="1512" alt="image" src="https://github.com/user-attachments/assets/f1ba02cf-fa29-4e9f-9ee2-208099a818d4">

I'm running the latest version on macOS 14.6.1 (23G93) installed through homebrew.

**Reply by ClementTsang** [reply]
Can you show me your config file? Or what arguments you're running it with?

**Reply by philocalyst** [reply]
Thank you for responding so quickly :)
```
[flags]
avg_cpu = true

# Temperature is one of:
temperature_type = "f"

rate = 1000
cpu_left_legend = false
current_usage = true
case_sensitive = false
whole_word = false
regex = true
default_widget_type = "proc"
default_widget_count = 1
tree = true
average_cpu_row = true

[processes]
# Pick which columns you want to use in any order.
columns = ["name", "cpu%", "mem%", "gpu%"]
[styles]
theme = "gruvbox"

[[row]]
  [[row.child]]
  type="cpu"
[[row]]
    ratio=2
    [[row.child]]
      ratio=4
      type="proc"
    [[row.child]]
      ratio=3
      [[row.child.child]]
        type="net"
      [[row.child.child]]
        type="batt"


```

**Reply by ClementTsang** [reply, marked as answer]
`current_usage = true` will make the reported process CPU% based on the _current_ total CPU usage, not its max capacity - so it can report process CPU as being much higher than what the CPU widget's average is reporting. Could you try setting that to false and letting me know if that gives you what you want?

**Reply by philocalyst** [reply]
Yes this was the right thing to do, thank you for being so committed to helping me out! Means the world to have such kind souls in the world.

**Reply by ClementTsang** [reply]
Glad you got it working! Also gave me some insight on some things to improve on with the config system and docs.

**Comment by philocalyst** [comment]
That's awesome! I did seriously look through the docs, I think the biggest 
issue with them currently is how configuration variables outside of a widget 
can impact that widget. I feel like a backlinks-like system could be effective 
when you're looking at the config options for a widget, or even just giving 
widgets associated with an option in its main description.

Just suggestions! Love the tool, definitely my favorite top clone (is that 
respectful?).
